lib/CMakeLists.txt

```diff
@@ -2,6 +2,7 @@ add_subdirectory(CAPI)
 add_subdirectory(Conversion)
 add_subdirectory(Dialect)
 add_subdirectory(Target)
+add_subdirectory(SharedLib)
 
 add_mlir_library(TTMLIR STATIC RegisterAll.cpp
```

Comment:
I'd rather we call `TTMLIR` -> `TTMLIRStatic` (note you'll have to update `ttmlir-opt` cmake which needs the static lib).  And let's call the shared lib just `TTMLIR`.

Reply:
I had trouble naming :) thanks for suggestion will do